pkg/render/dex_config.go

```diff
@@ -481,14 +481,17 @@ func (d *dexConfig) Connector() map[string]interface{} {
 
 	case connectorTypeGoogle:
 		config = map[string]interface{}{
-			"issuer":                 googleIssuer,
-			"clientID":               fmt.Sprintf("$%s", clientIDEnv),
-			"clientSecret":           fmt.Sprintf("$%s", clientSecretEnv),
-			"redirectURI":            fmt.Sprintf("%s/dex/callback", d.ManagerURI()),
-			"scopes":                 d.RequestedScopes(),
-			"serviceAccountFilePath": serviceAccountSecretLocation,
-			adminEmailSecretField:    fmt.Sprintf("$%s", googleAdminEmailEnv),
+			"issuer":       googleIssuer,
+			"clientID":     fmt.Sprintf("$%s", clientIDEnv),
+			"clientSecret": fmt.Sprintf("$%s", clientSecretEnv),
+			"redirectURI":  fmt.Sprintf("%s/dex/callback", d.ManagerURI()),
+			"scopes":       d.RequestedScopes(),
+		}
+		if d.idpSecret.Data[serviceAccountSecretField] != nil && d.idpSecret.Data[adminEmailSecretField] != nil {
+			config["serviceAccountFilePath"] = serviceAccountSecretLocation
```

Comment:
Minor: Same comment here [as the other version of this](https://github.com/tigera/operator/pull/1233/files#r597230453).